scheduler/compileserver.cpp

```diff
@@ -401,3 +410,139 @@ bool CompileServer::blacklisted(const Job *job, const pair<string, string> &envi
     Environments blacklist = job->submitter()->getEnvsForBlacklistedCS(this);
     return find(blacklist.begin(), blacklist.end(), environment) != blacklist.end();
 }
+
+int CompileServer::getInFd() const
+{
+    return m_inFd;
+}
+
+void CompileServer::setInFd(int fd)
+{
+    m_inFd = fd;
+}
+
+int CompileServer::startInConnectionTest()
+{
+    if (m_noRemote)
+    {
+        return -1;
+    }
+    else if (m_inFd != -1)
+    {
+        return m_inFd;
+    }
+    else if (m_nextConnTime > time(0))
+    {
+        return m_inFd;
+    }
+
+    m_inFd = socket(PF_INET, SOCK_STREAM, 0);
+    fcntl(m_inFd, F_SETFL, O_NONBLOCK);
+
+    struct hostent *host = gethostbyname(name.c_str());
+
+    struct sockaddr_in remote_addr;
+    remote_addr.sin_family = AF_INET;
+    remote_addr.sin_port = htons(remotePort());
+    memcpy(&remote_addr.sin_addr.s_addr, host->h_addr_list[0], host->h_length);
+    memset(remote_addr.sin_zero, '\0', sizeof(remote_addr.sin_zero));
+
+    int status = connect(m_inFd, (struct sockaddr *)&remote_addr, sizeof(remote_addr));
+    if ((status < 0) && (errno == EINPROGRESS || errno == EAGAIN))
+    {
+        m_lastConnStartTime=time(0);
+    }
+    else
```

Comment:
isn't there a status==0 case?  I know this is a non-blocking socket but we are not guaranteed that the connection doesn't finish that quick?